backend/src/layer/logic/mealplan_management/relation_resolver.rs

```diff
@@ -0,0 +1,254 @@
+use crate::util::Date;
+use std::slice::Iter;
+use tracing::warn;
+
+pub struct RelationResolver<DataAccess>
+where
+    DataAccess: MealplanManagementDataAccess + Send + Sync,
+{
+    db: DataAccess,
+}
+
+impl<DataAccess> RelationResolver<DataAccess>
+where
+    DataAccess: MealplanManagementDataAccess + Send + Sync,
+{
+    pub const fn _new(db: DataAccess) -> Self {
+        Self { db }
+    }
+
+    const PERCENTAGE: f64 = 0.8;
+    const fn get_edge_case_meal() -> &'static str {
+        "je 100 g"
+    }
+
+    /// This method resolves relation problems with canteen data and the corresponding database.<br>
+    /// After each resolve the object gets injected into the database.<br>
+    /// If an similar object already exists, the existing object will be updated with the new object data.<br>
+    /// `canteen: ParseCanteen`<br>This struct contains all canteen data e.g. lines and dishes.<br>
+    /// `date: Date`<br>This date decides when the meal will be served next.<br>
+    /// **Return**<br>Occurring errors get passed to the `MealPlanManger`.
+    pub async fn resolve(&self, canteen: ParseCanteen, date: Date) -> Result<(), DataError> {
+        match self.db.get_similar_canteen(&canteen.name).await? {
+            Some(similar_canteen) => {
+                self.db
+                    .update_canteen(similar_canteen.id, &canteen.name)
+                    .await?
+            }
+            None => self.db.insert_canteen(&canteen.name).await?,
+        };
+
+        for line in canteen.lines {
+            let name = &line.name.clone();
+            if let Err(_e) = self.handle_line(line, date).await {
+                warn!("Skip line '{:?}' as it could not be resolved", name);
+            }
+        }
+        Ok(())
+    }
+
+    async fn handle_line(&self, line: ParseLine, date: Date) -> Result<(), DataError> {
+        let db_line = match self.db.get_similar_line(&line.name).await? {
+            Some(similar_line) => self.db.update_line(similar_line.id, &line.name).await?,
+            None => self.db.insert_line(&line.name).await?,
+        };
+
+        let average = Self::determine_average_price(line.dishes.iter(), line.dishes.len())?;
+
+        for dish in line.dishes {
+            let name = &dish.name.clone();
+            if let Err(_e) = self.handle_dish(&db_line, dish, date, average).await {
+                warn!("Skip dish '{:?}' as it could not be resolved", name);
+            }
+        }
+        Ok(())
+    }
+
+    async fn handle_dish(
+        &self,
+        db_line: &Line,
+        dish: Dish,
+        date: Date,
+        average: u32,
+    ) -> Result<(), DataError> {
+        let similar_meal_result = self.db.get_similar_meal(&dish.name).await?;
+        let similar_side_result = self.db.get_similar_side(&dish.name).await?;
+        let price_limit = f64::from(average) * Self::PERCENTAGE;
+
+        // Case: A similar side and meal could be found. Uncommon case.
```

Comment:
what is with the first case?

Reply:
There could be the case, even it is uncommon, that the database finds a similar meal and side for a dish. This case is handled the same as just a similar meal could be found. 

So both cases are handled in this if block. 

=> if (meal is similar) & (side is similar) { dish is meal }